semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SymbolOps.scala

```diff
@@ -23,15 +23,20 @@ trait SymbolOps { self: DatabaseOps =>
           def definitelyLocal =
             sym == g.NoSymbol ||
               sym.name.decoded.startsWith(g.nme.LOCALDUMMY_PREFIX) ||
-              (sym.owner.isMethod && !sym.isParameter) ||
-              ((sym.owner.isAliasType || sym.owner.isAbstractType) && !sym.isParameter)
+              sym.owner.isMethod ||
+              sym.owner.isAliasType ||
+              sym.owner.isAbstractType
           !definitelyGlobal && (definitelyLocal || isLocal(sym.owner))
         }
         if (isLocal(sym)) {
           val mpos = sym.pos.toMeta
           return {
             if (mpos == m.Position.None) m.Symbol.None
-            else m.Symbol.Local(mpos.syntax)
+            else {
+              val id = idCache.get(mpos.input.syntax)
```

Comment:
General note, I've found it incredibly awkward to use `mpos.input.syntax` as a unique key. `Input` needs more work IMO

Reply:
Agreed. Opened https://github.com/scalameta/scalameta/issues/1240.